Support for Microchip Same51

danielinux opened the pull request:
- [x] hal support
- [x] Dual-bank swapping
- [x] update Target.md
- [x] Add build test

dgarske commented on .gdbinit:
```diff
@@ -1,6 +1,7 @@
 file wolfboot.elf
-add-symbol-file test-app/image.elf
+#add-symbol-file test-app/image.elf
+add-symbol-file /home/dan/src/wolfboot/IDE/MPLabX/wolfBoot_usbUpdateApp.X/dist/default/production/wolfBoot_usbUpdateApp.X.production.elf
```
Consider reverting this.

dgarske commented on IDE/MPLabX/wolfBoot-SAME51.X/Makefile:
```diff
@@ -0,0 +1,113 @@
+#
```
Please consider using `IDE/MPLABX` to match our other libraries and Microchip naming.

dgarske commented on hal/same51.c:
```diff
@@ -0,0 +1,389 @@
+void hal_init(void)
+{
+
+    uint32_t reg;
```
```hal/same51.c: In function 'hal_init':
hal/same51.c:280:14: error: unused variable 'reg' [-Werror=unused-variable]
  280 |     uint32_t reg;
      |              ^~~
```

dgarske commented on hal/same51.c:
```diff
@@ -0,0 +1,389 @@
+
+#include <string.h>
+
+static void RAMFUNCTION fork_bootloader(void)
```
Consider wrapping this in `DUALBANK_SWAP`.

```
hal/same51.c:263:25: error: 'fork_bootloader' defined but not used [-Werror=unused-function]
  263 | static void RAMFUNCTION fork_bootloader(void)
      |                         ^~~~~~~~~~~~~~~```

dgarske replied:
I added `CFLAGS+=-Wunused -Wunused-parameter` to get these.

dgarske commented on IDE/MPLabX/wolfBoot-SAME51.X/Makefile:
```diff
@@ -0,0 +1,113 @@
+#
+#  There exist several targets which are by default empty and which can be 
```
Please add a README.md for how to use these MPLABX projects.

danielinux commented:
Thanks for the review, @dgarske . I should have addressed the comments + re-created the two MPLAB projects + provided extended documentation. This is now ready to consider for merging.
